metals/src/main/scala/scala/meta/internal/metals/ServerCommands.scala

```diff
@@ -338,18 +339,19 @@ object ServerCommands {
     "[uri], the uri of the worksheet that you'd like to copy the contents of."
   )
 
-  val ExtractMemberDefinition = new Command(
-    "extract-member-definition",
-    "Extract member definition",
-    """|Whenever a user chooses a code action to extract a definition of a Class/Trait/Object/Enum this
-       |command is later ran to extract the code and create a new file with it
-       |""".stripMargin,
-    """|[uri, line, character], uri of the document that the command needs to be invoked on
-       |together with line number and character/column where the definition is.
-       |""".stripMargin
-  )
+  val ExtractMemberDefinition =
+    new ParametrizedCommand[TextDocumentPositionParams](
+      "extract-member-definition",
+      "Extract member definition",
+      """|Whenever a user chooses a code action to extract a definition of a Class/Trait/Object/Enum this
+         |command is later ran to extract the code and create a new file with it
+         |""".stripMargin,
+      """|[uri, line, character], uri of the document that the command needs to be invoked on
+         |together with line number and character/column where the definition is.
+         |""".stripMargin
+    )
 
-  val InsertInferredType = new Command(
+  val InsertInferredType = new ParametrizedCommand[TextDocumentPositionParams](
```

Comment:
I changed this one and `ExtractMemberDefinition` since this only used by the code actions (and shouldn't be done by the editor command), so we should be safe here.